js/main.js

```diff
@@ -33,4 +36,46 @@ function getRandomNumber(collectionLength) {
   return integer;
 }
 
+function handleClick(event) {
+  if (event.target.matches('.fa-solid')) {
+    var collectionData = {
+      collectionName: data.NFTvisible[0].collectionName,
+      collectionPhoto: null,
+      likes: null,
+      dislikes: null
+    };
+
+    var singleNFTData = {
+      name: data.NFTvisible[0].name,
+      data: data.NFTvisible[0].image
+    };
+
+    var xhr = new XMLHttpRequest();
+    xhr.open('GET', 'https://api.opensea.io/api/v1/collection/' + collectionData.collectionName.toLowerCase());
+    xhr.responseType = 'json';
+    xhr.addEventListener('load', function () {
+      collectionData.collectionPhoto = xhr.response.collection.image_url;
+    });
+    xhr.send();
+
+    if (Object.entries(data.ratingsInfo).length === 0) {
```

Comment:
What is this condition checking? Are you trying to see if this property already exists? If so, this would be a better way:
`if (data.ratingsInfo[collectionData.collectionName] === undefined){`

Reply:
got it